python/activity_stream/data_manager.py

```diff
@@ -560,34 +589,47 @@ def __db_insert_activity_updates(self, connection, cursor, entity_type, entity_i
         """
         self._bundle.log_debug("Updating database with %s new events" % len(events))
         try:
-            
             for event in events:
                 activity_id = event["id"]
-                
                 payload = cPickle.dumps(event, cPickle.HIGHEST_PROTOCOL)
                 blob = sqlite3.Binary(payload)
+
                 # first insert event
-                sql = """
-                    INSERT INTO activity(activity_id, payload, created_at) 
-                    SELECT ?, ?, datetime('now')
-                    WHERE NOT EXISTS(SELECT activity_id FROM activity WHERE activity_id = ?);                
-                 """
+                if self._force_activity_stream_update:
+                    sql = """
+                        INSERT INTO activity(activity_id, payload, created_at) 
```

Comment:
does this mean we are double inserting records? Should there be some stage where things are deleted from the db too?